### ISSUE TITLE
smoke test fail: delete dataset from user profile

### ISSUE THREAD

**Issue by only1chunts** [gigascience/gigadb-website#866]
**Describe the bug**
Does the delete dataset link on your datasets work? no.
It appears to at first, when it askes if youre sure you want to delete it, but when I clicked yes, it refreshed the page but the dataset was still listed in my table

**To Reproduce**
1. Go to https://staging.gigadb.org/user/view_profile#submitted
2. Click on "delete" button for any submitted datasets that do not have status "Published"
3. Click "OK" on pop-up confirmation
4. See ... nothing happen.

**Expected behavior**
once confirmed that you want to delete the dataset, it should be deleted from the database (along with directly and solely related items) and removed from your personal profile page.

**Screenshots**
![Screenshot_45](https://user-images.githubusercontent.com/6037145/144613282-8fb4eb14-a952-484c-828e-830cdaa9bd63.png)

**Additional context**
it appears something similar was reported before #357

This is a bug found during tests listed in #747

**Comment by pli888** [comment]
@only1chunts To confirm, when a user has deleted a dataset then this means that all metadata in the PostgreSQL database and all files belonging to the dataset need to be removed, not simply marked as deleted?

I think this ticket could do with some user stories so that we know what functionality to develop.

**Comment by only1chunts** [comment]
@pli888 , the current action is DELETE, so to replicate the current functionality we should do that. 
The Dataset table and dataset_file table are definately removed, as for all the other things related to the dataset I'm not sure how far the action goes! There are a number of IF statements that need to be accounted for with authors, samples and attributes so my guess would be that those are not deleted with the dataset. (e.g. IF an author is also used in another dataset then do not delete that author etc.)

In future we may wish to revisit the idea and add in an "undo" functionality which would require them to be marked for deletion instead of actually deleting them, so if the amount of work to mark instead of actually delete is about the same then it might be a future-time saving to do it now? One thing that might add to the work of that would be the ability to expunge those marked for deletion at some point in the future, say after 1 month.

**Comment by pli888** [comment]
@only1chunts - After discussion with @rija, we think the work involved with this ticket is too big to be part of the current Sprint 6. More planning on how to do this work is required because of the extra care needed when deleting data from the PostgreSQL database.

**Comment by only1chunts** [comment]
sounds reasonable. so as a quick hack to fix this bug can the "Delete" button (shown in the screenshot in ticket) be made to delete the username from the dataset table? that way the dataset will no longer show up in that users profile.
It maybe that the database has a constraint that username cannot be Null, if so make it add my username or that of a madeup Admin account.

**Comment by rija** [comment]
Yes, It's cleaner to associate it with a specially made-up user account when a user press delete for a dataset. 
That way, it also helps out with  retrieving the datasets that need deleting (i.e: all datasets owned by that special user)  when we get round to do it properly and fully.
